### PR TITLE
feat: align Java HTTP mapping with the new model and add a test

### PR DESCRIPTION
**Pull request by stuartwdouglas**

### CODE REVIEW

File: backend/controller/ingress/testdata/java/httpingress/src/main/java/xyz/block/ftl/java/test/http/Nested.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package xyz.block.ftl.java.test.http;
+
+import com.fasterxml.jackson.annotation.JsonAlias;
+
+public class Nested {
+
+    /**
+     * TODO: Jackson annotation support
+     */
+    @JsonAlias("good_stuff")
```
Comment (alecthomas) [inline backend/controller/ingress/testdata/java/httpingress/src/main/java/xyz/block/ftl/java/test/http/Nested.java:10]: !

File: backend/controller/ingress/testdata/java/httpingress/src/main/java/xyz/block/ftl/java/test/http/TestHTTP.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.POST;
+import jakarta.ws.rs.PUT;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.Produces;
+
+import jakarta.ws.rs.core.MediaType;
+import org.jboss.resteasy.reactive.ResponseHeader;
+import org.jboss.resteasy.reactive.ResponseStatus;
+import org.jboss.resteasy.reactive.RestPath;
```
Comment (alecthomas) [inline backend/controller/ingress/testdata/java/httpingress/src/main/java/xyz/block/ftl/java/test/http/TestHTTP.java:15]: Are these types extensions to the JAX-RS standard?

**Review by alecthomas — APPROVED** [review]

File: backend/controller/ingress/testdata/java/httpingress/src/main/java/xyz/block/ftl/java/test/http/TestHTTP.java (hunk 2 above)
Comment (stuartwdouglas) [reply backend/controller/ingress/testdata/java/httpingress/src/main/java/xyz/block/ftl/java/test/http/TestHTTP.java:15]: Yes. The Quarkus team wanted to try and push similar into the standard but I am not sure where that effort got to.

I will make sure to cover this when we talk about it in Sydney.

File: backend/controller/ingress/testdata/java/httpingress/src/main/java/xyz/block/ftl/java/test/http/Nested.java (hunk 1 above)
Comment (stuartwdouglas) [reply backend/controller/ingress/testdata/java/httpingress/src/main/java/xyz/block/ftl/java/test/http/Nested.java:7]: oops

**Pull request merged**